### PR TITLE
Update chemprop to v2.1

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
This PR will hold any changes for v2.1 until we are ready to release it. This allows us to continue to make minor changes and release a patch v2.0 version.. Any changes to API will be held until v2.1. This includes our uncertainty related changes.

### CODE REVIEW

**Comment by davidegraff** [comment]
do you mean "additions" to the API? Are we changing the API in a minor version bump?

**Comment by KnathanM** [comment]
We think we should not use batch normalization between the aggregation and predictor by default. Currently the default is to include a batch normalization layer. We found during benchmarking that in most cases batch normalization only helped a little and in some cases it made performance much worse, though we never looked into why this was the case. I believe this would be an API change (though model files do save whether batch_norm is True or False, this won't have compatibility issues). 

In terms of whether to make these kinds of changes in a minor version bump, let me know what you think. I'm still not too familiar  with the realm of software versioning.

**Comment by davidegraff** [comment]
I see. I guess it would be a breaking change, but in the name of generally better performance, I'm fine with this. These models don't really take long to train anyways.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ on:
   push:
     branches: [main]
   pull_request:
-    branches: [main]
+    branches: [main, v2-1/dev, v2/uncertainty]
```
Comment (shihchengli) [inline .github/workflows/ci.yml:15]: Remove this before merging
```suggestion
    branches: [main]
```

**Comment by davidegraff** [comment]
Why are we introducing a dependency on PyG/DGL in our testing suite? The core library does not rely on these, to my knowledge, so these tests would fail because of client code. This is not what we should be testing in our CI.

**Comment by davidegraff** [comment]
We can have whatever dependencies you want in client notebooks, but we need to be very thoughtful/careful about what our CI is testing. If you're testing client code and it fails a test, it can fail either because the core library is broken or you've written "bad" client code. The latter should not be a reason for our CI failing, but our tests can't distinguish that so they should be _excluded_ from our CI.
